samza-core/src/main/java/org/apache/samza/operators/StreamGraphImpl.java

```diff
@@ -116,38 +127,31 @@ public StreamGraph withContextManager(ContextManager contextManager) {
    *
    * @param streamName the name of the stream to be created. Will be prefixed with job name and id to generate the
    *                   logical streamId.
-   * @param keyExtractor the {@link Function} to extract the outgoing key from the intermediate message
-   * @param msgExtractor the {@link Function} to extract the outgoing message from the intermediate message
-   * @param msgBuilder the {@link BiFunction} to convert the incoming key and message to a message
-   *                   in the intermediate {@link MessageStream}
-   * @param <K> the type of key in the intermediate message
-   * @param <V> the type of message in the intermediate message
+   * @param serde the {@link Serde} to use for the message in the intermediate stream. If null, the default serde
+   *              is used.
```

Comment:
Minor: can add `{@link #setDefaultSerde}` to provide a default serde

Reply:
This is a protected method called by MessageStreamImpl (i.e. us), so we shouldn't be setting the default serde before calling. Hence this is only documenting that the default serde (whether set or the NoOp) will be used. Will add the link if we ever expose this in public API.